docs/reference/filesystems/gcs.md

```diff
@@ -0,0 +1,86 @@
+---
+title: GCS
+---
+
+# Google Cloud Storage (GCS) File System
+
+The Google Cloud Storage (GCS) file system allows GlareDB to read data directly from Google Cloud Storage buckets.
+This makes it easy to query data stored in cloud-based data lakes without
+needing to download files manually.
+
+The GCS file system is enabled by default for the CLI, Python bindings, and
+WebAssembly bindings.
+
+## Usage
+
+To read from a GCS bucket, use any supported file-reading function with a
+`gs://` URI:
+
+```sql
+SELECT * FROM read_parquet('gs://my-bucket/data.parquet');
+```
+
+This reads the `data.parquet` file from the `my-bucket` bucket.
+
+You can also query GCS files directly by specifying the GCS URI in the FROM clause:
+
+```sql
+SELECT * FROM 'gs://my-bucket/data.parquet';
+```
+
+## Optional Parameters
+
+When accessing private GCS buckets, you can provide a service account key directly as
+an option:
+
+```sql
+SELECT * FROM read_parquet(
+  'gs://my-private-bucket/secure.parquet',
+  service_account = '{SERVICE_ACCOUNT_JSON}'
+);
+```
+
+### Supported Options
+
+| Option            | Description                                                                 |
+|-------------------|-----------------------------------------------------------------------------|
+| `service_account` | GCP Service Account key in JSON format                                      |
+
+## Supported Formats
+
+All supported file formats, such as CSV and Parquet, can be read using the GCS
+file system using their respective functions (e.g. `read_parquet`).
+
+## CORS Configuration for WebAssembly
+
+When accessing GCS buckets from GlareDB running in a browser using WebAssembly,
+you need to configure Cross-Origin Resource Sharing (CORS) for your GCS bucket to
+allow browser-based requests.
+
+To configure CORS for your GCS bucket:
+
+1. Navigate to your GCS bucket in the Google Cloud Console
+2. Go to **Permissions** -> **CORS configuration**
+3. Add the following JSON configuration:
+
+```json
+[
+    {
+        "origin": ["*"],
+        "method": ["GET", "HEAD"],
+        "responseHeader": ["Content-Type"],
+        "maxAgeSeconds": 3600
+    }
+]
+```
```

Comment:
This can't be done in the console.

Steps needed:

1. Create a json file (cors.json) with the following configuration:
```json
[
  {
    "origin": ["*"],
    "method": ["GET", "HEAD"],
    "responseHeader": ["*"],
    "maxAgeSeconds": 3600
  }
]
```

2. Apply the configuration using gsutil:
```
gsutil cors set cors.json gs://<my-bucket>
```